app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -401,9 +401,7 @@ protected void onPause() {
             exitImmersiveSync();
         }
         mAudioEngine.pauseEngine();
-
-        mWindows.onPause();
```

Comment:
- `saveState()` is run inside windows.onPause(), are we missing that call now?

Reply:
I'm also concerned that if a window is hidden the mWindow.onPause not getting called from the loop because it may have been removed from the mWidgets container:
   ```
     for (Widget widget: mWidgets.values()) {
            widget.onPause();
        }
```

Reply:
Updated so we call `saveState`.

The onPause/onResume calls are also called for hidden widgets so they should be called for any existing window.